src/compiler/symb.bas

```diff
@@ -2017,32 +2017,26 @@ private function hSymbCheckConstAssignFuncPtr _
 			rsubtype->typ, rsubtype->subtype )
 	if( match = FB_OVLPROC_NO_MATCH ) then
 		wrnmsg = FB_WARNINGMSG_RETURNTYPEMISMATCH
-		exit function
-	end if
 
 	'' Does one have a BYREF result, but not the other?
-	if( symbIsRef( lsubtype ) <> symbIsRef( rsubtype ) ) then
+	elseif( symbIsRef( lsubtype ) <> symbIsRef( rsubtype ) ) then
 		wrnmsg = FB_WARNINGMSG_RETURNMETHODMISMATCH
-		exit function
-	end if
 
 	'' Different calling convention?
-	if( symbAreProcModesEqual( lsubtype, rsubtype ) = FALSE ) then
+	elseif( symbAreProcModesEqual( lsubtype, rsubtype ) = FALSE ) then
 		wrnmsg = FB_WARNINGMSG_CALLINGCONVMISMATCH
-		exit function
```

Comment:
Wouldn't removing the early exits change the behaviour, i.e. it returns TRUE now (as long as the loop checking the parameters doesn't detect any problem)?

Reply:
Yea.  The intent was return both parts separately:
TRUE/FALSE - the consts match or not
wrnmsg=0, <>0 - the function pointer args, callconv, etc, match or not

I don't think this quite right.  Doen't make much sense to checks consts, if the function ptr is fundamentally mismatched.